addons/mrp/mrp.py

```diff
@@ -541,6 +550,11 @@ def _mrp_from_move(self, cr, uid, ids, context=None):
             res += self.pool.get("mrp.production").search(cr, uid, [('move_lines', 'in', move.id)], context=context)
         return res
 
+    def _get_mo_ids(self, cr, uid, product_template_id, context=None):
+        # It will return Manufacturing Orders ids of product.
+        product_ids = self.pool['product.product'].search(cr, uid, [('product_tmpl_id', 'in', product_template_id)], context=context)
+        return self.pool["mrp.production"].search(cr, uid , [('product_id', 'in', product_ids)], context=context)
```

Comment:
you can search like [('product_id.product_tmpl_id', 'in', product_template_id)]

so no need to right two search statements